adapters/repos/db/vector/hnsw/startup.go

```diff
@@ -197,6 +203,44 @@ func (h *hnsw) restoreFromDisk() error {
 	return nil
 }
 
+func (h *hnsw) restoreDocMappings() error {
+	prevDocID := uint64(0)
+	relativeID := uint64(0)
+	maxNodeID := uint64(0)
+	for _, node := range h.nodes {
+		if node == nil {
+			continue
+		}
+		nodeIDBytes := make([]byte, 8)
+		binary.BigEndian.PutUint64(nodeIDBytes, node.id)
+		docIDBytes, err := h.store.Bucket(h.id + "_mv_mappings").Get(nodeIDBytes)
+		if err != nil {
+			return errors.Wrap(err, "get docIDBytes")
+		}
+		docID := binary.BigEndian.Uint64(docIDBytes)
+		if docID != prevDocID {
+			relativeID = 0
+			prevDocID = docID
+		}
+		if h.compressed.Load() {
```

Comment:
What happens if a compression happens while we're restoring the mappings?
Would some of it be in the compressor cache and some be in the other cache?